### PR TITLE
Bug 2019972: Update CSR model & fix approval/denial process

### PR DESCRIPTION
**Pull request by rawagner**

### CODE REVIEW

File: frontend/packages/metal3-plugin/src/selectors/csr.ts [hunk 1]
```diff
@@ -38,22 +37,31 @@ export const getNodeClientCSRs = (csrs: CertificateSigningRequestKind[] = []): C
...
-  return Object.keys(groupped).map((key) => {
+  return Object.keys(groupped).reduce((acc, key) => {
     const { csr, status } = groupped[key][0];
-    return {
-      metadata: { name: key },
-      status,
-      csr,
-    };
-  });
+    if (isCSRPending(csr)) {
```
Comment (spadgett) [inline frontend/packages/metal3-plugin/src/selectors/csr.ts:56]: This is fine, but it's more performant to filter before sorting. Any reason for this change?

Comment (rawagner) [reply frontend/packages/metal3-plugin/src/selectors/csr.ts:56]: Yes, i need to decode CSR `spec.request` first to get the name of node. With the decoded name I can tell which CSRs belong to which node and look only at the latest request.

I think that in the past when CSR got approved/denied then all old CSRs were approved/denied too, but that is not happening now. So I changed the frontend code to consider only the latest CSR for every node.

Comment (rawagner) [reply frontend/packages/metal3-plugin/src/selectors/csr.ts:56]: I was trying to find out why the old CSRs are not auto-approved/denied but could not find out. I think it was the case in the past but Im not sure.

In any case I got confirmed twice that looking at latest CSR is enough as all older ones will be garbage collected after 24hrs.

**Review by jtomasek — APPROVED** [review]
/lgtm

**Pull request merged**